app/models/manageiq/providers/cloud_manager/virtual_template.rb

```diff
@@ -0,0 +1,14 @@
+class ManageIQ::Providers::CloudManager::VirtualTemplate < ::MiqTemplate
+  validate :single_template, :on => :create
+  default_value_for :cloud, true
+
+  TYPES = {
+    :amazon => 'ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate'
+  }.freeze
+
+  def single_template
+    multiple = !type.constantize.where(:type => type).empty?
```

Comment:
Can you explain what this does?  Typically you shouldn't have to query on `:type` directly because ActiveRecord's STI layer will do that for you.  Also, the type column already represents `self`'s type, so `type.constantize` is the same as `self.class`.  If you're worried about new records not yet having a subclass, we have a mixin we use that simplifies that called NewWithTypeStiMixin (which you already should have since it was inherited from MiqTemplate).